### ISSUE TITLE
[WAF - Reliability Review] - avm/res/api-management/service

### ISSUE THREAD

**Issue by jtracey93** [Azure/bicep-registry-modules#1184]
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Feature Request

### Module Name

avm/res/api-management/service

### Description

Hello @Azure/avm-res-apimanagement-service-module-owners-bicep,

The AVM core team are performing a review of all AVM modules to ensure they are aligned to the WAF reliability pillar by default, and in the future they will be checking against other WAF pillars also. This is a requirement we have before we can go `v1.0.0` for any AVM module.

Links to AVM specs:

- https://azure.github.io/Azure-Verified-Modules/specs/shared/#id-sfr2---category-composition---waf-aligned
- https://azure.github.io/Azure-Verified-Modules/specs/shared/#id-sfr5---category-composition---availability-zones

Links to WAF Reliability & APRL recommendations used for analysis:

- https://learn.microsoft.com/azure/well-architected/reliability/checklist
- https://learn.microsoft.com/azure/well-architected/service-guides/api-management/reliability
- https://azure.github.io/Azure-Proactive-Resiliency-Library/services/integration/api-management/

Whilst reviewing this module the following issues have been found that are non-compliant with the AVM specs that need to be resolved in a new version release:

- [ ] The default SKU should be set to `Premium` so AZs can be enabled - https://github.com/Azure/bicep-registry-modules/blob/main/avm/res/api-management/service/main.bicep#L57-L65
- [ ]  The resource should be spread across all 3 AZs by default - https://github.com/Azure/bicep-registry-modules/blob/main/avm/res/api-management/service/main.bicep#L91-L93

Please make these changes in a PR to the parameter defaults so a new minor version can be published.

Thanks

The AVM Core Team

**Comment by jtracey93** [comment]
superseded by #1473
